net/node/link.go

```diff
@@ -44,8 +44,14 @@ func unpackNodeBuf(node *node, buf []byte) {
 			errors.New("Unexpected size of received message")
 			return
 		}
-		// FIXME Check the payload < 0 error case
-		msgLen = PayloadLen(buf) + MSGHDRLEN
+		length := 0
+
+		length, buf = PayloadLen(buf)
+
+		if length == 0 && buf == nil {
+			return
```

Comment:
should return error to the caller. otherwise function node.rx will keep on calling unpackNodeBuf， which 
obviously will get error since the buf has been consumed in this call.